File: system-http-client/src/main/java/com/levnovikov/core_network/converter/GsonConverter.java
package com.levnovikov.core_network.converter;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.levnovikov.core_network.api_provider.EntityConverter;

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

public class GsonConverter implements EntityConverter {

    private Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd").create();

    @Override
    public <T> String convertTo(T entity) {
        return gson.toJson(entity);
    }

    @Override
    public <T> T convertFrom(String string, Class<T> responseClass) {
        return gson.fromJson(string, responseClass);
    }
}

File: core-network/src/main/java/com/levnovikov/core_network/converter/GsonConverter.kt
package com.levnovikov.core_network.converter

import com.levnovikov.core_network.api_provider.EntityConverter
import kotlin.reflect.KClass

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

class GsonConverter : EntityConverter {


    override fun <T> convertTo(entity: T): String {
        TODO()
    }

    override fun <T : Any> convertFrom(string: String, responseClass: KClass<T>): T {
        TODO()
    }
}
